src/p11_child/p11_child_openssl.c

```diff
@@ -383,20 +382,16 @@ static errno_t do_ocsp(struct p11_ctx *p11_ctx, X509 *cert)
 
     OCSP_request_add1_nonce(ocsp_req, NULL, -1);
 
-    if (p11_ctx->timeout != -1  && p11_ctx->cert_verify_opts->soft_ocsp) {
-        /* decrease timeout of time so far spend in p11_child */
-        /* substract 1 so we finish before child is forcibly terminated */
-        req_timeout = p11_ctx->timeout - 1 -
-            (time(NULL) - p11_ctx->start_timestamp);
-        if (req_timeout < 0) {
+    if (p11_ctx->ocsp_deadline != -1  && p11_ctx->cert_verify_opts->soft_ocsp) {
+        req_timeout = p11_ctx->ocsp_deadline - time(NULL);
```

Comment:
There is a `difftime()`.
But it returns `double`.
I now tend to agree to keep everything as time_t and that's it.

Reply:
done